pyam/core.py

```diff
@@ -83,6 +86,19 @@ def __init__(self, data, **kwargs):
         if 'exec' in run_control():
             self._execute_run_control()
 
+    def _format_data_time_col(self):
```

Comment:
I don't like adding two functions with almost entirely the same name. I think we have two options:

- make a single function that can be inherited by subclasses that must handle both `'year'` and `'time'`
- make two functions, one explicitly for year and one explicitly for time

Reply:
Or, appending to option first: move default behavior for year out and only have one function for time

Reply:
Thinking of it more, I like option two better. But honestly would prefer if we could get rid of `'year'` entirely...

Reply:
I think that a substantial part of our users and use cases will still revolve around yearly data, so I'd very much like to retain the option to do `df.filter(year=2020)`. Plus we should cater to the use case of having categorical sub-annual specifications like `['summer-day', 'winter-night']`.

If this can be maintained while removing `year` on the backend, fine by me.

Reply:
Ok will go for option two. I don't see a way to get rid of year and match the sub-annual specifications yet so will leave that in for now and leave that upgrade to you as I think you know the use cases better.